code/modules/reagents/reagent_dispenser.dm

```diff
@@ -65,6 +65,12 @@
 	icon_state = "fuel"
 	reagent_id = "welding_fuel"
 
+/obj/structure/reagent_dispensers/fueltank/attackby(obj/item/I, mob/user, params, prevent_warning = 0)
+	if((istype(I, /obj/item/weapon/weldingtool) || /obj/item/weapon/weldingtool/largetank || /obj/item/weapon/weldingtool/largetank/cyborg || /obj/item/weapon/weldingtool/mini || /obj/item/weapon/weldingtool/hugetank || /obj/item/weapon/weldingtool/experimental || /obj/item/weapon/weldingtool/experimental/brass) || I.welding = 1

```

Comment:
istype(I, /obj/item/weapon/weldingtool) returns true for all children of weldingtool, no need to check each version of it.
You need to typecast I as a weldingtool before you can check the welding var: after the istype, `var/obj/item/weapon/weldingtool/W = I`  and then `if(W.welding)`.